enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -383,24 +388,43 @@ type Subnet struct {
 // +kubebuilder:validation:Pattern=`^subnet-[0-9A-Za-z]+$`
 type AWSSubnetID string
 
-// SubnetRole specifies the roles (aka functions) that the subnet will provide in the cluster.
-type SubnetRole string
+// SubnetRole specifies the role (aka functions) that the subnet will provide in the cluster.
+// +kubebuilder:validation:XValidation:rule=`self.role != 'ControlPlaneLB' || has(self.controlPlaneLB)`,message="controlPlaneLB is required when role is ControlPlaneLB"
+// +kubebuilder:validation:XValidation:rule=`self.role != 'ControlPlaneLB' || has(self.ingressControllerLB)`,message="controlPlaneLB is required when role is IngressControllerLB"
+type SubnetRole struct {
+    // type specifies the type of role (aka function)
+    // that the subnet will provide in the cluster.
```

Comment:
It would be good to include the list of available roles within this godoc (I assume we generate docs from the types here as well?)

Reply:
Right, done.